src/ch/blinkenlights/android/vanilla/LibraryPagerAdapter.java

```diff
@@ -852,6 +851,38 @@ public void onItemClick (AdapterView<?> parent, View view, int position, long id
 	}
 
 	/**
+	 * Perform usability-related actions on pager and contained lists, e.g. highlight current song
+	 * or scroll to it if opted-in
+	 * @param song song that is currently playing, can be null
+     */
```

Comment:
It's iterating over view children and does mainly UI-related actions, are you sure it would be best place for it? The complexity is now O(n) and I'm planning to maintain it at this level.

Otherwise it would look like 
```java
// background thread handler
handleMessage(Message msg) {
    switch(msg.what) {
    ...
        case SONG_CHANGE_REQUESTED:
            <needed stuff here>
            mActivity.runOnUiThread(new Runnable {
                <some ugly code just to touch views>
            })
            break;
    ...
    }
}
```